tests/virtualization/universal/hotplugging.pm

```diff
@@ -20,122 +20,204 @@ use utils;
+    my $domblk_target = 'vdz';
+    $domblk_target = 'xvdz' if (is_xen_host);
+    script_run("virsh detach-disk $guest ${domblk_target}", 240);
+    if (try_attach("virsh attach-disk --domain $_ --source $disk_image --target ${domblk_target}")) {
+        assert_script_run "virsh domblklist $guest | grep ${domblk_target}";
+        # Skip lsblk check for VIRT_AUTOTEST KVM test suites after attaching raw disk due to uncertainty
+        if (!get_var('VIRT_AUTOTEST') && is_kvm_host) {
+            my $lsblk = script_run("ssh root\@$guest lsblk | grep va[b-z]", 60);
             record_soft_failure("lsblk failed - please check the output manually") if $lsblk != 0;
-            $self->{test_results}->{$guest}->{"lsblk failed - please check the output manually (ssh root\@$guest lsblk | grep ${domblk_target})"}->{status} = 'SOFTFAILED' if (get_var('VIRT_AUTOTEST') && ($lsblk != 0));
         }
+        assert_script_run("ssh root\@$guest lsblk");
+        assert_script_run("virsh detach-disk $guest ${domblk_target}", 240);
     }
-    assert_script_run "ssh root\@$_ lsblk" foreach (keys %virt_autotest::common::guests);
-    assert_script_run "virsh detach-disk $_ ${domblk_target}", 240 foreach (keys %virt_autotest::common::guests);
-
-    # TODO:
-    record_info "CPU", "Changing the number of CPUs available";
-    foreach my $guest (keys %virt_autotest::common::guests) {
-        if (get_var('VIRT_AUTOTEST') && is_kvm_host && ($sles_running_version eq '11' && $sles_running_sp eq '4')) {
-            record_info 'Skip vCPU hotplugging', 'bsc#1169065 vCPU hotplugging does no work on SLE 11-SP4 KVM host';
-            next;
-        }
-        unless ($guest =~ m/hvm/i) {
-            # The guest should have 2 CPUs after the installation
-            assert_script_run "virsh vcpucount $guest | grep current | grep live", 90;
-            assert_script_run "ssh root\@$guest nproc",                            60;
-            # Add 1 CPU for everu guest
-            assert_script_run "virsh setvcpus --domain $guest --count 3 --live";
-            sleep 5;
-            assert_script_run "virsh vcpucount $guest | grep current | grep live | grep 3";
-            if (get_var('VIRT_AUTOTEST') && is_kvm_host && ($sles_running_version eq '15' && $sles_running_sp eq '2')) {
-                my $vcpu_nproc = script_retry "ssh root\@$guest nproc", delay => 60, retry => 3, timeout => 60, die => 0;
-                record_soft_failure 'bsc#1170026 vCPU hotplugging damages this guest ' . $guest                                      if ($vcpu_nproc != 0);
-                $self->{test_results}->{$guest}->{"bsc#1170026 vCPU hotplugging damages this guest $guest"}->{status} = 'SOFTFAILED' if ($vcpu_nproc != 0);
-            } else {
-                script_retry "ssh root\@$guest nproc", delay => 60, retry => 3, timeout => 60;
-            }
-        }
+    assert_script_run("rm -f $disk_image");
+}
+
+# Add a virtual CPU to the given guest
+sub add_vcpu {
```

Comment:
Don't you want to do `set_vcpus()` instead?
In this subroutine we add vpu and then we remove it.

I suggest something like:
```perl
set_vcpus($_, 3);
set_vcpus($_, 2);
```

Reply:
I've updated `hotplugging.pm` accordingly